Common/Net/HTTPNaettRequest.cpp

```diff
@@ -0,0 +1,96 @@
+#ifndef HTTPS_NOT_AVAILABLE
+
+#include <cstring>
+
+#include "Common/Net/HTTPRequest.h"
+#include "Common/Net/HTTPNaettRequest.h"
+#include "Common/Thread/ThreadUtil.h"
+#include "Common/Log.h"
+
+#include "ext/naett/naett.h"
+
+namespace http {
+
+HTTPSDownload::HTTPSDownload(RequestMethod method, const std::string &url, const std::string &postData, const std::string &postMime, const Path &outfile, ProgressBarMode progressBarMode, const std::string &name)
+	: Download(url, name, &cancelled_), method_(method), postData_(postData), postMime_(postMime), outfile_(outfile), progressBarMode_(progressBarMode) {
+}
+
+HTTPSDownload::~HTTPSDownload() {
+	Join();
+}
+
+void HTTPSDownload::Start() {
+	_dbg_assert_(!req_);
+
+	const char *methodStr = method_ == RequestMethod::GET ? "GET" : "POST";
+	req_ = naettRequest_va(url_.c_str(), !postMime_.empty() ? 3 : 2, naettMethod(methodStr), naettHeader("accept", "application/json, text/*; q=0.9, */*; q=0.8"), naettHeader("Content-Type", postMime_.c_str()));
+	res_ = naettMake(req_);
+}
+
+void HTTPSDownload::Join() {
+	if (!res_ || !req_)
+		return;  // No pending operation.
+	// Tear down.
+	if (completed_ && res_) {
+		_dbg_assert_(req_);
+		naettClose(res_);
+		naettFree(req_);
+		res_ = nullptr;
+		req_ = nullptr;
+	} else {
+		ERROR_LOG(IO, "HTTPSDownload::Join not implemented");
+	}
+}
+
+bool HTTPSDownload::Done() {
+	if (completed_)
+		return true;
+
+	if (naettComplete(res_)) {
+		resultCode_ = naettGetStatus(res_);
```

Comment:
I guess if naett doesn't support cancelling requests.  Maybe we should at least check that flag to make sure `ResultCode()` returns something appropriate is set (as I think it would be in our current) in the response when it's read.

We could just do like `IsCancelled()` here:
```suggestion
		resultCode_ = IsCancelled() ? -6 : naettGetStatus(res_);
```

That would just make sure we don't have any surprising changes.

-[Unknown]